pkg/operator/ceph/cluster/controller.go

```diff
@@ -192,6 +211,22 @@ func (c *ClusterController) GetClusterCount() int {
 // ************************************************************************************************
 // Add event functions
 // ************************************************************************************************
+func (c *ClusterController) operatorConfigAdd(obj interface{}) {
+	cm, ok := obj.(*v1.ConfigMap)
+	if !ok {
+		logger.Warningf("Expected ConfigMap but handler received %T. %#v", obj, obj)
+		return
+	}
+
+	logger.Infof("ConfigMap %q changes detected. Updating configurations", cm.Name)
+	for _, callback := range c.operatorConfigCallbacks {
+		if err := callback(); err != nil {
+			logger.Errorf("%v", err)
+		}
+	}
```

Comment:
@travisn @leseb Do we need to `c.csiConfigMutex.Lock()` and `c.csiConfigMutex.Unlock()` around this?

Reply:
Instead of adding the lock around these callbacks, I think we need to move the locks inside the `updateDrivers()` method. Let's worry about that as a separate PR though.